src/core/libmaven/mzSample.cpp

```diff
@@ -116,20 +116,20 @@ string mzSample::getFileName(const string& filename)
     return ("");
 }
 
-void mzSample::loadAnySample(const char* filename)
+void mzSample::loadAnySample(string filename)
 {
-    if (mystrcasestr(filename, "mzCSV") != NULL) {
-        parseMzCSV(filename);
-    } else if (mystrcasestr(filename, "mzdata") != NULL) {
-        parseMzData(filename);
-    } else if (mystrcasestr(filename, "mzxml") != NULL) {
-        parseMzXML(filename);
-    } else if (mystrcasestr(filename, "mzml") != NULL) {
-        parseMzML(filename);
-    } else if (mystrcasestr(filename, "cdf") != NULL) {
-        parseCDF(filename, 1);
+    if (contains(filename, "mzCSV") == true) {
+        parseMzCSV(filename.c_str());
+    } else if (contains(filename, "mzdata") == true ) {
+        parseMzData(filename.c_str());
+    } else if (contains(filename, "mzxml") == true) {
+        parseMzXML(filename.c_str());
+    } else if (contains(filename, "mzml") == true) {
+        parseMzML(filename.c_str());
+    } else if (contains(filename, "cdf") == true) {
+        parseCDF(filename.c_str(), 1);
```

Comment:
The expression `contains(string, subString)` itself evaluates to a boolean primitive and would satisfy the conditional. You need not check its result against `true` - that is redundant logic. In fact it is more readable, without being checked against true.